packages/experience/src/pages/SocialSignInWebCallback/use-social-sign-in-listener.ts

```diff
@@ -184,20 +203,37 @@ const useSocialSignInListener = (connectorId: string) => {
 
     const { state, ...rest } = parseQueryParameters(searchParameters);
 
-    const isGoogleOneTap = validateGoogleOneTapCsrfToken(
-      rest[GoogleConnector.oneTapParams.csrfToken]
-    );
+    // Google One Tap always contains the `credential`
+    const isGoogleOneTap = isGoogleOneTapChecker(rest);
+    // External Google One Tap always contains the `credential` and doesn't contain the `csrfToken`
+    // Experience built-in Google One Tap always contains the `csrfToken`
+    const isExternalCredential = isExternalGoogleOneTapChecker(rest);
 
     // Cleanup the search parameters once it's consumed
     setSearchParameters({}, { replace: true });
 
-    if (!validateState(state, connectorId) && !isGoogleOneTap) {
+    const isValidAuth = getAuthValidationResult({
+      isGoogleOneTap,
+      state,
+      connectorId,
+      isExternalCredential,
+      params: rest,
+    });
+
+    if (!isValidAuth) {
       setToast(t('error.invalid_connector_auth'));
       navigate('/' + experience.routes.signIn);
       return;
     }
 
-    if (!verificationIdRef.current && !isGoogleOneTap) {
+    const isValidSession = getSessionValidationResult({
+      verificationId: verificationIdRef.current,
+      isGoogleOneTap,
+      isExternalCredential,
+      params: rest,
+    });
+
+    if (!isValidSession) {
```

Comment:
nit: Can move these to a validation hook. E.g.  `const { validateSocialCallback } = useSocialCallbackValidation();`

simply run `validateSocialCallback()` here;